pkg/component/dependencywatchdog/access.go

```diff
@@ -58,6 +73,22 @@ func NewAccess(
 	}
 }
 
+// NewDWDAccess creates a new instance of dependencyWatchdogAccess for shoot cluster access for the dependency-watchdog.
+// TODO(aaronfern): remove this when g/g v1.91 is released
+func NewDWDAccess(
+	client client.Client,
+	namespace string,
+	secretsManager secretsmanager.Interface,
+	values AccessValues,
+) *dependencyWatchdogAccess {
+	return &dependencyWatchdogAccess{
+		client:         client,
+		namespace:      namespace,
+		secretsManager: secretsManager,
+		values:         values,
+	}
+}
```

Comment:
What is the difference of this function compared to `NewAccess`?

Reply:
`NewAccess` returns the `Deployer` interface in contrast to `NewDWDAccess` returning the `dependencyWatchdogAccess` struct. We could not invoke the `DeployMigrate()` method from `NewAccess`

Reply:
Wouldn't it be an option to change `NewAccess` to return *dependencyWatchdogAccess? It still implements `component.Deployer` and should not have downsides?

Reply:
Yes, definitely as option
I just did not think of it. Thanks for pointing it out, will be able to get rid of `NewDWDAccess` :)